packages/web3-validator/src/utils.ts

```diff
@@ -149,11 +149,19 @@ export const abiSchemaToJsonSchema = (
 			abiType = abi;
 			abiName = `${level}/${index}`;
 
-			// If its provided in short form of tuple e.g. [['uint', 'string']]
+			// If it's provided in short form of tuple e.g. [['uint', 'string']]
 		} else if (Array.isArray(abi)) {
 			// If its custom tuple e.g. ['tuple[2]', ['uint', 'string']]
-			if (abi[1] && Array.isArray(abi[1])) {
-				abiType = abi[0] as string;
+			if (
+				abi[0] &&
+				typeof abi[0] === 'string' &&
+				String(abi[0]).startsWith('tuple') &&
```

Comment:
```suggestion
			abi[0].startsWith('tuple') &&
```
I dont think casting to string is neccesary

Reply:
this one need to be sure the method startsWith exists